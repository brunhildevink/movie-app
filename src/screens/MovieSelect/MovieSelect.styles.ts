import styled from 'styled-components'
import { colors } from '../../styles'

const EpisodeSelection = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: 200px;
  grid-gap: 0 32px;
  overflow-x: scroll;
  padding-left: 40px;
`

const Container = styled.div`
  max-width: 1280px;
  width: 100%;
  padding: 40px;
  display: flex;
  align-items: center;
  margin-left: auto;
  margin-right: auto;
`

const Wrapper = styled.div<{ img: string }>`
  display: inline-grid;
  color: ${colors.white};
  height: 100%;
  background: linear-gradient(236deg, rgba(31, 0, 19, 0.444437149859944) 0%, rgba(215, 0, 133, 0.217546393557423) 100%),
    ${({ img }) => `url(${img})`} no-repeat center center;
  background-blend-mode: screen;
  background-size: cover;
  height: 100%;
  width: 100%;
  grid-template-rows: 60% 40%;
`

export { Container, EpisodeSelection, Wrapper }
