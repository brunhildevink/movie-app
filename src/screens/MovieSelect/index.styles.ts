import styled from 'styled-components'
import { breakpoints, colors, fonts, typography } from '../../styles'

const ButtonContainer = styled.div`
  display: flex;
  align-self: flex-end;
  margin: 20px 0;

  @media ${breakpoints.lg} {
    margin: 20px 24px;
  }
`

const EpisodeSelection = styled.div`
  display: flex;
  flex-direction: column;
  overflow-x: hidden;
  padding: 40px;

  @media ${breakpoints.lg} {
    padding: 0 0 0 40px;
  }
`

const Container = styled.div`
  max-width: 1280px;
  width: 100%;
  padding: 40px;
  display: flex;
  align-items: center;
  margin-left: auto;
  margin-right: auto;
`

const Select = styled.select`
  font-family: ${fonts};
  background-color: transparent;
  color: ${colors.white};
  border: none;
  font-size: ${typography.fontSizeHeadingThree}px;
  line-height: ${typography.lineHeightHeadingThree};
`

const Wrapper = styled.div<{ img: string }>`
  position: relative;
  display: inline-grid;
  color: ${colors.white};
  height: 100%;
  background: ${colors.white};
  background: linear-gradient(236deg, rgba(187, 0, 115, 0.5) 0%, rgba(0, 51, 134, 0.5)), ${({ img }) => `url(${img})`};
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
  width: 100%;

  @media ${breakpoints.lg} {
    grid-template-rows: 60% 40%;
  }

  .swiper-container {
    display: flex;
    width: 100%;
  }

  .swiper-wrapper {
    width: 100%;
  }

  .swiper-slide {
    display: block;
    width: 100%;
  }

  @media ${breakpoints.md} {
    .swiper-slide {
      display: block;
      width: 200px !important;
    }
  }
`

export { ButtonContainer, Container, EpisodeSelection, Select, Wrapper }
