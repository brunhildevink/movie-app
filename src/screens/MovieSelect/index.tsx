import React from 'react'
import { Container, EpisodeSelection, Wrapper } from './MovieSelect.styles'
import image from '../../assets/images/1.1.jpg'
import { Heading, Text } from '../../components'
import Card from '../../components/Card'
import { Episode } from '../../lib/types'

interface Props {
  episodes: Episode[]
  title: string
  totalSeasons: number
  onSelect: (season: number) => void
}

const MovieSelect: React.FC<Props> = ({ episodes, title, totalSeasons, onSelect }) => {
  const renderSelectSeasons = Array.from(Array(totalSeasons).keys()).map((num) => (
    <option key={num} value={num + 1}>
      Season {num + 1}
    </option>
  ))

  const renderCards = episodes.map((episode) => <Card key={episode.imdbID} id={episode.imdbID} />)

  return (
    <Wrapper img={image}>
      <Container>
        <div>
          <select
            key={totalSeasons}
            disabled={!(totalSeasons > 0)}
            onChange={(event) => onSelect(parseInt(event.target.value, 10))}
          >
            {renderSelectSeasons}
          </select>
          <Heading.HeadingOne>{title}</Heading.HeadingOne>
          <Text.Regular>
            Follows the awkward experiences and racy tribulations of a modern-day mandalorian.
          </Text.Regular>
        </div>
      </Container>
      <EpisodeSelection>{renderCards}</EpisodeSelection>
    </Wrapper>
  )
}

export default MovieSelect
