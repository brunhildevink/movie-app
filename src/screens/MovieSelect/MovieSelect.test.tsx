import React from 'react'
import renderer from 'react-test-renderer'
import MovieSelect from '.'

const mocks = [
  {
    Title: 'Chapter 1: The Mandalorian',
    Released: '2019-11-12',
    Episode: '1',
    imdbRating: '8.6',
    imdbID: 'tt9095424',
  },
  {
    Title: 'Chapter 2: The Child',
    Released: '2019-11-15',
    Episode: '2',
    imdbRating: '8.6',
    imdbID: 'tt9121530',
  },
  {
    Title: 'Chapter 3: The Sin',
    Released: '2019-11-22',
    Episode: '3',
    imdbRating: '9.0',
    imdbID: 'tt9121534',
  },
  {
    Title: 'Chapter 4: Sanctuary',
    Released: '2019-11-29',
    Episode: '4',
    imdbRating: '7.8',
    imdbID: 'tt9121536',
  },
  {
    Title: 'Chapter 5: The Gunslinger',
    Released: '2019-12-06',
    Episode: '5',
    imdbRating: 'N/A',
    imdbID: 'tt9121538',
  },
  {
    Title: 'Chapter 6: The Prisoner',
    Released: '2019-12-13',
    Episode: '6',
    imdbRating: '8.3',
    imdbID: 'tt9121542',
  },
  {
    Title: 'Chapter 7: The Reckoning',
    Released: '2019-12-18',
    Episode: '7',
    imdbRating: '9.0',
    imdbID: 'tt9121544',
  },
]

test('MovieSelect renders correctly', () => {
  const tree = renderer
    .create(<MovieSelect description="hllo" episodes={mocks} title="Hello" totalSeasons={8} onSelect={() => null} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
