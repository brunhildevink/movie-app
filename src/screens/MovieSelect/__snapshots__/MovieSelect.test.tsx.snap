// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSelect renders correctly 1`] = `
<div
  className="sc-dkzDqf ddXhXL"
>
  <div
    className="sc-gsnTZi BqpBJ"
  >
    <div>
      <select
        disabled={false}
        onChange={[Function]}
      >
        <option
          value={1}
        >
          Season 
          1
        </option>
        <option
          value={2}
        >
          Season 
          2
        </option>
        <option
          value={3}
        >
          Season 
          3
        </option>
        <option
          value={4}
        >
          Season 
          4
        </option>
        <option
          value={5}
        >
          Season 
          5
        </option>
        <option
          value={6}
        >
          Season 
          6
        </option>
        <option
          value={7}
        >
          Season 
          7
        </option>
        <option
          value={8}
        >
          Season 
          8
        </option>
      </select>
      <h1
        className="sc-hKMtZM cqdelg"
      >
        Hello
      </h1>
      <p
        className="sc-kDDrLX cdZjdm"
      >
        hllo
      </p>
    </div>
  </div>
  <div
    className="sc-bczRLJ hTOvVw"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
